endaq/calc/psd.py

```diff
@@ -225,6 +225,51 @@ def to_octave(
     return result
 
 
+def loglog_linear_approx(
+    df: pd.DataFrame,
+    knots: typing.List[int],
+    window: int = 1,
+    freqs_out: typing.Union[typing.Literal["knots", "input"], np.ndarray] = "knots",
+):
+    """
+    Calculate a piecewise-linear approximation of data, in the log-log space.
+
+    :param df: the input data; each column is approximated independently
+    :param knots: the inner boundary points separating the piecewise regions
+    :apram window: the length of the pre-processing smoothing window
+    :param freqs_out: the frequencies at which to interpolate the output; these
+        may be explicit frequencies, or one of the following `str` options:
```

Comment:
It might be good to be a little more information about the shape and contents of the explicit frequencies, but that might be postponed until we do a larger documentation/example sprint.

Reply:
good call; I put a small change in 4a3fa05c2d53e2744daecfb76a934071ec8367d3 which is hopefully good enough for now, but I'm up for revisiting this in the docs push